### PR TITLE
add mmocr demo documentation

### PR DESCRIPTION
**Pull request by quincylin1**
add mmocr demo documentation for the following:
- overall readme in /demo
- readme for textdet and textrecog in /docs
- images in /resources

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Demo
+
+This page provides tutorials about running demos. Please click the caption for more information.
+
+<br>
+
+<div align="center">
+    <img src="../resources/demo_text_det_pred.jpg"/><br>
+
+[Text Detection](docs/text_det_demo.md)
```
Comment (jeffreykuang) [inline demo/README.md:10]: This link is not correct

File: demo/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+</div>
+
+<br>
+
+<div align="center">
+    <img src="../resources/demo_text_recog_pred.jpg" width="200px" alt/><br>
+
+[Text Recognition](docs/text_recog_demo.md)
```
Comment (jeffreykuang) [inline demo/README.md:18]: This link is not correct. Pls click the caption online

File: docs/text_det_demo.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Take [PANet](https://download.openmmlab.com/mmocr/textdet/panet/panet_r18_fpem_ffm_sbn_600e_icdar2015_20210219-42dbe46a.pth) (pre-trained with ICDAR2015 for 600 epochs) as an example:
+
+```shell
+python demo/image_demo.py ${TEST_IMG} ${CONFIG_FILE} ${CHECKPOINT_FILE} ${SAVE_PATH} [--imshow] [--device ${GPU_ID}]
+```
+
+Example:
+
+```shell
+python demo/image_demo.py demo/demo_text_det.jpg configs/textdet/panet/panet_r18_fpem_ffm_600e_icdar2015.py https://download.openmmlab.com/mmocr/textdet/panet/panet_r18_fpem_ffm_sbn_600e_icdar2015_20210219-42dbe46a.pth demo/demo_text_det_pred.jpg
```
Comment (jeffreykuang) [inline docs/text_det_demo.md:19]: are you sure our script can automatically download pth from web?

File: docs/text_det_demo.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+The predicted result will be saved as `demo/demo_text_det_pred.jpg`.
```
Comment (jeffreykuang) [inline docs/text_det_demo.md:22]:
```suggestion
The predicted result will be saved as `demo/demo_text_det_pred.jpg.
```

File: docs/text_recog_demo.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+python demo/image_demo.py ${TEST_IMG} ${CONFIG_FILE} ${CHECKPOINT_FILE} ${SAVE_PATH} [--imshow] [--device ${GPU_ID}]
+```
+
+Example:
+
+```shell
+python demo/image_demo.py demo/demo_text_recog.jpg configs/textrecog/sar/sar_r31_parallel_decoder_academic.py https://download.openmmlab.com/mmocr/textrecog/sar/sar_r31_parallel_decoder_academic-dba3a4a3.pth demo/demo_text_recog_pred.jpg
+```
+
+The predicted result will be saved as `demo/demo_text_recog_pred.jpg`.
```
Comment (jeffreykuang) [inline docs/text_recog_demo.md:22]:
```suggestion
The predicted result will be saved as `demo/demo_text_recog_pred.jpg.
```

**Review by jeffreykuang — COMMENTED** [review]
pls rm demo_text-det_pred.jpg and demo_text_recog_pred.jpg

**Review by cuhk-hbsun — APPROVED** [review]

**Pull request merged**